src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1175,7 +1175,27 @@ size_t QgsOgrProvider::layerCount() const
  */
 QgsWkbTypes::Type QgsOgrProvider::wkbType() const
 {
-  return static_cast<QgsWkbTypes::Type>( mOGRGeomType );
+  OGRwkbGeometryType wkb = mOGRGeomType;
+
+  if ( ogrDriverName == QLatin1String( "ESRI Shapefile" ) )
+  {
+    switch ( mOGRGeomType )
```

Comment:
Would a simple `QgsWkbTypes::multiType( mOGRGeomType );` work as well? Are there other types that need to be reported as single types?